scripts/commit-message

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+set -e
+
+fail() {
+  (echo && sed 's/^/! /' <<<"$1" && echo) >&2
+  exit 1
+}
+
+# Find the number of the most recently opened pull request merged so far.
+# Add 1 to this to get the smallest possible number of the next pull request,
+# then add an arbitrary buffer to allow for issues and pull requests opened
+# since the creation of the most recently opened pull request merged so far.
```

Comment:
A suggestion to make the purpose of script more clear:

```diff
# Add 1 to this to get the smallest possible number of the next pull request,
# then add an arbitrary buffer to allow for issues and pull requests opened
# since the creation of the most recently opened pull request merged so far.
+# This gives an estimation of the number that will appear in the merge commit
+# message, allowing us to assert its length.
```

And possibly extract the part which actually does this, so that the comment relates directly to what follows:

```sh
# <the_comment>
pr_number_estimation=$(git log --grep ...)

message="... $pr_number_estimation ..."
```